control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1659,25 +1697,66 @@ func (r *reconciler) reconcileObservedConfiguration(ctx context.Context, hcp *hy
 }
 
 func (r *reconciler) reconcileCloudConfig(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
-	// This is needed for the e2e tests and only for Azure: https://github.com/openshift/origin/blob/625733dd1ce7ebf40c3dd0abd693f7bb54f2d580/test/extended/util/cluster/cluster.go#L186
-	if hcp.Spec.Platform.Type != hyperv1.AzurePlatform {
-		return nil
-	}
 
-	reference := cpomanifests.AzureProviderConfig(hcp.Namespace)
-	if err := r.cpClient.Get(ctx, client.ObjectKeyFromObject(reference), reference); err != nil {
-		return fmt.Errorf("failed to fetch %s/%s configmap from management cluster: %w", reference.Namespace, reference.Name, err)
-	}
+	switch hcp.Spec.Platform.Type {
+	case hyperv1.AzurePlatform:
+		// This is needed for the e2e tests and only for Azure: https://github.com/openshift/origin/blob/625733dd1ce7ebf40c3dd0abd693f7bb54f2d580/test/extended/util/cluster/cluster.go#L186
+		reference := cpomanifests.AzureProviderConfig(hcp.Namespace)
+		if err := r.cpClient.Get(ctx, client.ObjectKeyFromObject(reference), reference); err != nil {
+			return fmt.Errorf("failed to fetch %s/%s configmap from management cluster: %w", reference.Namespace, reference.Name, err)
+		}
 
-	cm := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: "openshift-config", Name: "cloud-provider-config"}}
-	if _, err := r.CreateOrUpdate(ctx, r.client, cm, func() error {
-		if cm.Data == nil {
-			cm.Data = map[string]string{}
+		cm := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: "openshift-config", Name: "cloud-provider-config"}}
```

Comment:
Can you make the name of the namespace and the config map constants?
The same for line 1745.